### DISCUSSION TITLE
How do I build query in SQLAlchemy Core with parameters being labeled as $1, $2, $3, etc.?

### DISCUSSION THREAD

**Discussion by MarkBrook** [sqlalchemy/sqlalchemy#9276] [Usage Questions] [answered]
Hi! Right now when I'm building a query with some filtering compiled query is resulting in something like:
```
SELECT * FROM table1 WHERE name = :name_1 OR name = :name_2
```

How can I transform it to be like this:
```
SELECT * FROM table1 WHERE name = $1 OR name = $2
```

**Comment by CaselIT** [comment, marked as answer]
Hi,

You would need to specify what dialect you want yo use when compiling the query

**Reply by MarkBrook** [reply]
It's postgres

**Reply by CaselIT** [reply]
then you would have to create an instance of the dialect for the postgresql dbapi you are using

**Reply by MarkBrook** [reply]
Yeah but it still compiles to query with a named parameters (:name_1, :name_2, etc. ) placeholders, not like $1, $2, ...

This is needed for me to use asyncpg parameters as it does not support the syntax above.

**Reply by zzzeek** [reply]
Use the asyncpg dialect to compile

```py

>>> from sqlalchemy import select, column
>>> from sqlalchemy import create_engine
>>> stmt = select(column('q')).where(column('q') == 5).where(column('p') == 17)
>>> e = create_engine("postgresql+asyncpg://")
>>> print(stmt.compile(e))
SELECT q 
WHERE q = $1::INTEGER AND p = $2::INTEGER


```

or make the regular PG dialect and set paramstyle to "numeric_dollar"

```py
>>> from sqlalchemy.dialects import postgresql
>>> d = postgresql.dialect(paramstyle="numeric_dollar")
>>> print(stmt.compile(dialect=d))
SELECT q 
WHERE q = $1 AND p = $2
```

**Reply by CaselIT** [reply]
> Yeah but it still compiles to query with a named parameters (:name_1, :name_2, etc. ) placeholders, not like $1, $2, ...
> 
> This is needed for me to use asyncpg parameters as it does not support the syntax above.

I'm not sure why you need to get the sql string, can't you just execute it via sqlalchemy?
